server/embed/serve.go

```diff
@@ -109,63 +111,87 @@ func (sctx *serveCtx) serve(
 
 	sctx.lg.Info("ready to serve client requests")
 
-	m := cmux.New(sctx.l)
+	var m cmux.CMux
+	var server func() error
+	onlyGRPC := splitHttp && !sctx.httpOnly
+	grpcEnabled := !splitHttp || !sctx.httpOnly
```

Comment:
It's a little hard to understand, suggest to change the logic to 
```
onlyHttp := sctx.httpOnly
onlyGRPC := splitHttp && !sctx.httpOnly

if onlyHttp {

} else if onlyGRPC {

} else {
   // support legacy behavior: both http and gRPC share the same conection
}
```

Reply:
Will try it, we will see if that's more readable.